libs/community/langchain_community/embeddings/octoai_embeddings.py

```diff
@@ -1,100 +1,86 @@
-from typing import Any, Dict, List, Mapping, Optional
+from typing import Dict
 
-from langchain_core.embeddings import Embeddings
-from langchain_core.pydantic_v1 import BaseModel, Extra, Field, root_validator
-from langchain_core.utils import get_from_dict_or_env
+from langchain_core.pydantic_v1 import Field, SecretStr, root_validator
+from langchain_core.utils import convert_to_secret_str, get_from_dict_or_env
 
-DEFAULT_EMBED_INSTRUCTION = "Represent this input: "
-DEFAULT_QUERY_INSTRUCTION = "Represent the question for retrieving similar documents: "
+from langchain_community.embeddings.openai import OpenAIEmbeddings
+from langchain_community.utils.openai import is_openai_v1
 
+DEFAULT_API_BASE = "https://text.octoai.run/v1/"
+DEFAULT_MODEL = "thenlper/gte-large"
 
-class OctoAIEmbeddings(BaseModel, Embeddings):
+
+class OctoAIEmbeddings(OpenAIEmbeddings):
```

Comment:
will this error if openai_api_key isn't passed in / OPENAI_API_KEY isn't set?

Reply:
No. It requires `OCTOAI_API_TOKEN` only.